### ISSUE TITLE
u32/u64 <--> int32/int64: dangerous to allow?

### ISSUE THREAD

**Issue by dannywillems** [zshipko/ocaml-rs#130]
The doc says u32 (resp. u64) is mapped to int32 (resp. u64). Is it not dangerous to allow this regarding overflow? Should it be mapped to a non native Caml value like done (though using an indirection) in [ocaml-integers](https://github.com/yallop/ocaml-integers)?

**Comment by zshipko** [comment]
Yeah, i think this is a situation where we've chosen to prioritize usability at the expense of some safety. In cases where you are concerned about overflow it makes sense to create a custom type that holds the Rust `u64`.
